topdown/eval.go

```diff
@@ -2892,6 +2913,42 @@ func (e evalTerm) save(iter unifyIterator) error {
 
 }
 
+type evalEvery struct {
+	e         *eval
+	generator *ast.Expr
+	body      ast.Body
+}
+
+func (e evalEvery) eval(iter unifyIterator) error {
+	child := e.e.closure(ast.NewBody(e.generator))
+	all := true // all generator evaluations yield one successful body evaluation
+	err := child.Run(func(child *eval) error {
```

Comment:
I was wondering about tracing for the closure. I guess the `Run()` call takes care of that.

Reply:
That's what I thought at least. We can adjust this -- I'm not sure, for example, about extra traces for the body evals.